libs/providers/flagd/src/lib/service/grpc/service.ts

```diff
@@ -8,21 +9,24 @@ import {
 } from '@openfeature/nodejs-sdk';
 import { Struct } from '../../../proto/ts/google/protobuf/struct';
 import { ServiceClient } from '../../../proto/ts/schema/v1/schema.client';
-import { Service } from '../Service';
+import { Service } from '../service';
+import { Protocol } from './protocol';
 
 interface GRPCServiceOptions {
   host: string;
   port: number;
+  protocol: Protocol
 }
+
 export class GRPCService implements Service {
   client: ServiceClient;
 
-  constructor(options: GRPCServiceOptions) {
-    const { host, port } = options;
-    this.client = new ServiceClient(
+  constructor(options: GRPCServiceOptions, client?: ServiceClient) {
```

Comment:
Again allowing injection of `ServiceClient` for easy testing (we can inject a mock)